README.md

```diff
@@ -830,6 +756,73 @@ JSON printer and parser can be generated using the --json flag or
+       SCHEMA_FILES <schema-file> [<schema-file> [...]]
+       [OUTPUT_DIR <output-directory>]
+       [ALL] [BINARY_SCHEMA] [COMMON] [COMMON_READER] [COMMON_BUILDER] [BUILDER]
+       [READER] [VERIFIER] [JSON_PARSER] [JSON_PRINTER] [JSON] [RECURSIVE]
+       [OUTFILE <output-file>] [PREFIX <prefix>] [TARGET <target-name>]
+       [PATHS <include-path> [<include-path> [...]]]
+       [EXTRA_ARGS <flatcc-flag> [<flatcc-flag> [...]]]
+    )
+
+This function sets the following targets and variables:
+
+- `flatcc_generated::<name>` target: linking to this target will automatically
+call flatcc for re-generating the sources when the input definition files
+have changed
+
+- `<name>_GENERATED_SOURCES` variable: list of all sources that are generated by flatcc.
+These can be used by adding they to `add_library`/`add_executable`.
+
+- `flatcc_generated_<name>` target: this target can be used as target of your
+make program to re-generate the flatcc headers when an input source has changed.
+
+This function accepts the following arguments:
+
+- NAME `<name>` is a unique name on which all output variables/targets are based.
+
+- `SCHEMA_FILES` is a required argument. As argument, it expects
+a list of flatbuffer definition files. These sources must be available when
+running this function as they are parsed for dependencies. If the definition files
+are generated dynamically, then you should make use
+[`add_custom_command`](https://cmake.org/cmake/help/latest/command/add_custom_command.html).
+See [test/monster_test_solo] for an example on how to do this.
+
+- `ALL`, `BINARY_SCHEMA`, `COMMON`, `COMMON_READER`, `COMMON_BUILDER`, `BUILDER`,
+`READER`, `VERIFIER`, `JSON_PARSER`, `JSON_PRINTER` and `JSON` are boolean
+options that instruct flatcc what types of source/binary to generate.
+Documentation of all arguments can be found at the top of
+`FlatccGEnerateSources.cmake`.
+
+This function uses CMake's `add_custom command` internally. This means that
+the generated headers are only available in the build step, not the configure
+step.
+
+Example:
+
+    find_package(flatcc REQUIRED COMPONENTS cli runtime)
+
+    include(FlatccGenerateSources)
+    flatcc_generate_sources(
+        NAME seclif_protocol
+        SCHEMA_FILES "datadef/seclif_protocol.fbs"
+        OUTPUT_DIR "${CMAKE_CURRENT_BINARY_DIR}/datadef"
+        BUILDER
+        VERIFIER
+    )
+
+    add_executable(my_main main.c)
+    target_link_libraries(seclif_protocol PRIVATE flatcc::runtime flatcc_generated::seclif_protocol)
+
+Cross building with flatbuffers on cmake is possible by providing the install location of the build architecture flatcc package via environment variable flatccCli_ROOT or via cmake variable flatccCli_ROOT. The flatcc_generate_sources function will then use the *build* architecture flatcc cli executable to generate the header files and link your target to the *host* architecture libflatccrt library. See section [Cross-compilation](#cross-compilation) for an example.
+
```

Comment:
suggest adding section:

Note: normally COMMON should also be specified along with BUILDER unless common files are generated by another build step. For simple configurations ALL is likely the best choice.

Reply:
Ok, I will add that fragment.